http/logical_test.go

```diff
@@ -242,5 +242,5 @@ func TestLogical_RequestSizeLimit(t *testing.T) {
 	resp := testHttpPut(t, token, addr+"/v1/secret/foo", map[string]interface{}{
 		"data": make([]byte, MaxRequestSize),
 	})
-	testResponseStatus(t, resp, 413)
+	testResponseStatus(t, resp, 400)
```

Comment:
I'm torn on this. We could use `errwrap.Contains` to check for the specific message and convert the status to 413, rather than fork MaxBytesReader. I don't know why Go doesn't return 413 in this case in the first place, but probably just an oversight. At this point they won't change it, although maybe they can be convinced to make it a public typed error to make catching it easier.

The problem of course is that then you have to actually ensure that you do this all the time, anywhere in Vault where an error is at the http level. That is more complicated than it might seem. So let's leave at 400 for now and we can make it "more correct" later on.